content/blog/2024/12/how-to-migrate-from-idf-bootloader-to-mcuboot-using-esp-self-reflasher/index.md

```diff
@@ -0,0 +1,271 @@
+| -------------------------------- | --------- | -------- |
+| Bootloader                       | 0x001000  | 0x00F000 |
+| Application primary slot         | 0x010000  | 0x100000 |
+| Application secondary slot       | 0x110000  | 0x100000 |
+| Scratch                          | 0x210000  | 0x040000 |
+
+
+---
+**Note**
+
+If MCUboot is being build separately from NuttX build, be sure that the configuration for flash organization matches on both.
+
+---
+
+Save and exit the menuconfig.
+
+Build the MCUboot bootloader:
+
+```bash
+make -j bootloader
+```
+
+Now build the NuttX application:
+
+```bash
+make ESPTOOL_BINDIR=./ -s -j
+```
+
+### Building the Reflashing application
+
+#### Example 1: Download the Target Reflashing Image
+
+ESP Self-Reflasher component provides the example `boot_swap_download_example`, this can be used as start point for the Reflashing application.
+
+```bash
+cd <IDF_DIR>/components/esp-self-reflasher/examples/boot_swap_download_example/
+```
+
+First, copy the *final target images* (NuttX and MCUboot binaries) to the directory that will be served for HTTP download.
+
+```bash
+cp <NUTTX_DIR>/nuttx.bin <IDF_DIR>/components/esp-self-reflasher/examples/boot_swap_download_example/example_bin_dir/app_upd.bin
+cp <NUTTX_DIR>/mcuboot-esp32.bin <IDF_DIR>/components/esp-self-reflasher/examples/boot_swap_download_example/example_bin_dir/mcuboot-esp32.bin
+```
+
+Configure the example:
+
+```bash
+idf.py menuconfig
+```
+
+Enter menu `Example Configuration -> Bootloader reflash configuration` and set the `Bootloader bin url endpoint` with the host IP, port and the bootloader binary name (e.g. `http://192.168.0.100:8070/mcuboot-esp32.bin`):
+
+```
+http://<HOST_IP>:<HOST_PORT>/mcuboot-esp32.bin
+```
+
+Also configure the `Bootloader region address` (the Bootloader offset mentioned above) and the `Bootloader region size`.
+
+Then configure the similar parameters on the `Example Configuration -> Application reflash configuration` menu, set the `Application upgrade url endpoint` (e.g. `http://192.168.0.100:8070/app_upd.bin`):
+
+```
+http://<HOST_IP>:<HOST_PORT>/app_upd.bin
+```
+
+Set the `Application region address` and `Application region size`. Note that if the `boot_swap_download_example` is flashed standalone directly as is, these must match the image Secondary Slot from the target MCUboot image organization as the example expects MCUboot to do the swapping to the Primary Slot later, otherwise, if the example is OTA upgraded which this tutorial walk through, these configs must match Primary Slot. It will depend on where the reflashing application is running from (Factory partition or OTA partition), as it should not overwrite itself on the process.
+
+Configure the preferred network connection settings on the menu `Example Connection Configuration`, such as Wi-Fi SSID and Password.
+
+Save and exit the menuconfig.
+
+Build the example:
+
+```bash
+idf.py build
+```
+
+#### Example 2: Embedded the Target Reflashing Image
```

Comment:
Embedded > Embed?

The same here, looks like this section has a procedure or a how-to, please format it as a procedure with distinct numbered steps and indented code blocks to show that they belong to this particular step.